instrumentation/netty/netty-4-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/v4/common/NettyFutureInstrumentation.java

```diff
@@ -86,36 +92,37 @@ public static void wrapListener(
           wrappedListeners[i] = listeners[i];
         }
       }
-      listeners = wrappedListeners;
+      return wrappedListeners;
     }
   }
 
   @SuppressWarnings("unused")
   public static class RemoveListenerAdvice {
 
     @Advice.OnMethodEnter
-    public static void wrapListener(
-        @Advice.Argument(value = 0, readOnly = false)
-            GenericFutureListener<? extends Future<?>> listener) {
-      listener = FutureListenerWrappers.getWrapper(listener);
+    @Advice.AssignReturned.ToArguments(@ToArgument(value = 0, typing = Assigner.Typing.DYNAMIC))
+    public static Object wrapListener(
+        @Advice.Argument(value = 0) GenericFutureListener<? extends Future<?>> listener) {
+      return FutureListenerWrappers.getWrapper(listener);
     }
   }
 
   @SuppressWarnings("unused")
   public static class RemoveListenersAdvice {
 
     @Advice.OnMethodEnter
-    public static void wrapListener(
-        @Advice.Argument(value = 0, readOnly = false)
-            GenericFutureListener<? extends Future<?>>[] listeners) {
+    @Advice.AssignReturned.AsScalar
```

Comment:
[for reviewer] here the `AsScalar` allows to assign the value of the returned array to the argument value, otherwise it's considered to be an `Object[]` that contains the arguments/return value/thrown exception assignments that bytebuddy has to do after the advice is invoked.

Reply:
can you add this as a code comment? thanks